### PR TITLE
Implement a validity function using JSON schemas

### PR DESCRIPTION
**Pull request by MichaelKim20**
1. I implemented the automatic creation of the interfaces using JSON Schemas.
    - This was solved using the library package `json-schema-to-typescript`
    - Whenever 'npm start' and 'npm test' are executed, they are automatically generated.
2. I Implemented a validity function using JSON schemas and interfaces
    - This was solved using the library package `ajv`.

Related to #55

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,9 +23,11 @@
   "dependencies": {
     "@types/sodium-native": "^2.3.5",
     "@types/sqlite3": "^3.1.6",
+    "ajv": "^6.12.3",
     "assert": "^2.0.0",
     "body-parser": "^1.19.0",
     "express": "^4.17.1",
+    "json-schema-to-typescript": "^9.1.1",
```
Comment (MichaelKim20) [inline package.json:30]: This is a library that creates an interface using json-schema.

File: package.json [hunk 2]
```diff
@@ -22,9 +23,11 @@
   "dependencies": {
     "@types/sodium-native": "^2.3.5",
     "@types/sqlite3": "^3.1.6",
+    "ajv": "^6.12.3",
```
Comment (MichaelKim20) [inline package.json:26]: This is a library that creates a validity function using json-schema and interface.

**Review by MichaelKim20 — COMMENTED** [review]
I removed the automatically generated file.
You can see the code here.
https://gist.github.com/MichaelKim20/764cc481a5148c9fbf6bb299ad4ce067

File: src/modules/data/schemas/Transaction.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    },
+    "outputs": {
+      "items": {
+        "type": "object"
+      },
+      "type": "array"
+    }
+  },
+  "additionalProperties": false,
+  "required": ["type", "inputs", "outputs"]
```
Comment (TrustHenry) [inline src/modules/data/schemas/Transaction.json:22]: What happens if there are no inputs like Genesis Block?

File: src/modules/data/validator/index.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+/**
+ * Class for validating JSON data
+ */
+export class Validator
+{
+    /**
+     * The map of validation functions created to reuse
+     * a once created validation function.
```
Comment (TrustHenry) [inline src/modules/data/validator/index.ts]:
```suggestion
     * An once created validation function.
```

File: tests/ValidationJSON.test.ts [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        assert.ok(Validator.isValidOtherwiseThrow<ITransaction>("Transaction", {
+            "type": 1,
+            "inputs": [],
+            "outputs": [
+                {
+                    "value": "400000000000",
+                    "address": "GA3DMXTREDC4AIUTHRFIXCKWKF7BDIXRWM2KLV74OPK2OKDM2VJ235GN"
+                }
+            ]
+        }));
```
Comment (TrustHenry) [inline tests/ValidationJSON.test.ts]: Input value is required on normal case .

**Review by TrustHenry — APPROVED** [review]
Nice! , Just small nits

File: src/modules/data/schemas/Transaction.json (hunk 3 above)
Comment (MichaelKim20) [reply src/modules/data/schemas/Transaction.json:22]: That's a good question.
In the absence of the content, it consists of:
```
{
...
    "inputs": [],
...
}
```

Comment (TrustHenry) [reply src/modules/data/schemas/Transaction.json:22]: When validation, Does it not get caught on `required` items?

File: src/modules/data/validator/index.ts (hunk 4 above)
Comment (MichaelKim20) [reply src/modules/data/validator/index.ts]:
```
The map of validation functions created to reuse -
an once created validation function.
```
I'll change it like this. Do you like it?

File: src/modules/data/schemas/Transaction.json (hunk 3 above)
Comment (MichaelKim20) [reply src/modules/data/schemas/Transaction.json:22]: The transactions in the Genesis block have no elements in the inputs.
The transaction is being handled normally.

File: tests/ValidationJSON.test.ts (hunk 5 above)
Comment (MichaelKim20) [reply tests/ValidationJSON.test.ts]:
``` typescript
    {
            "type": 0,
            "inputs": [
                {
                    "previous": "0x5d7f6a7a30f7ff591c8649f61eb8a35d034824ed5" +
                        "cd252c2c6f10cdbd2236713dc369ef2a44b62ba113814a9d819" +
                        "a276ff61582874c9aee9c98efa2aa1f10d73",
                    "index": 1,
                    "signature": "0x07557ce0845a7ccbba61643b95e310bd3ae06c41" +
                        "fab9e8761ff3b0e5d28a5d625a3b951223c618910b239e7b779" +
                        "c6c671252a78edff4d0f37bdb25982e4f4228"
                }
            ],
            "outputs": [
                {
                    "value": "400000000000",
                    "address": "GA3DMXTREDC4AIUTHRFIXCKWKF7BDIXRWM2KLV74OPK2" +
                        "OKDM2VJ235GN"
                }
            ]
        }
```

File: src/modules/data/schemas/Transaction.json (hunk 3 above)
Comment (MichaelKim20) [reply src/modules/data/schemas/Transaction.json:22]: The test code below is also being processed normally.
``` TypeScript
        assert.ok(Validator.isValidOtherwiseThrow<ITransaction>
        ("Transaction",
        {
            "type": 0,
            "inputs": [],
            "outputs": [
                {
                    "value": "400000000000",
                    "address": "GA3DMXTREDC4AIUTHRFIXCKWKF7BDIXRWM2KLV74OPK2" +
                        "OKDM2VJ235GN"
                }
            ]
        }));
```

Comment (MichaelKim20) [reply src/modules/data/schemas/Transaction.json:22]: The previous test code was also left.

**Pull request merged**